File: Algorithms/p137_Single_Number_II.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int a = 0;
        int b = 0;
        for(int v : nums){
            int b_n = (b&~v)|(a&v);
            int a_n = (a&~v)|(~b&~a&v);
            b = b_n;
            a = a_n;
        }
        return a;
    }
};

File: Algorithms/p137_Single_Number_II/p137_Single_Number_II.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        a = 0
        b = 0
        for v in nums:
            b_n = (b&~v)|(a&v)
            a_n = (a&~v)|(~b&~a&v)
            b = b_n
            a = a_n
        return a
